marlowe-playground-client/webpack.config.js

```diff
@@ -1,145 +1,141 @@
+                                ".spago/*/*/src/**/*.purs",
+                                "web-common/**/*.purs",
+                                "web-common-marlowe/**/*.purs",
+                                "web-common-playground/**/*.purs",
                             ],
-                            psc: null,
-                            bundle: !(isWebpackDevServer || isWatch),
-                            warnings: true,
-                            watch: isWebpackDevServer || isWatch,
-                            pscPackage: false,
-                            pscIde: false
-                        }
-                    }
-                ]
+                            psc: "psa",
+                            spage: true,
+                            bundle: !isDevelopment,
+                            watch: isDevelopment,
+                        },
+                    },
+                ],
             }, {
                 test: /\.tsx?$/,
-                loader: "ts-loader"
+                loader: "ts-loader",
             },
             {
                 test: /\.css$/,
-                use: ['style-loader', 'css-loader']
+                use: [MiniCssExtractPlugin.loader, "css-loader"],
             },
             {
                 test: /\.scss$/,
-                use: ['style-loader', 'css-loader', 'sass-loader']
+                use: [MiniCssExtractPlugin.loader, "css-loader", "sass-loader"],
             },
             {
-                test: /\.(gif|png|jpe?g|svg)$/i,
-                use: 'url-loader'
+                test: /\.(png|svg|jpg|jpeg|gif)$/i,
+                type: "asset/resource",
             },
             {
-                test: /\.ttf$/,
-                use: ['file-loader'],
-            }
-        ]
+                test: /\.(woff|woff2|eot|ttf|otf)$/i,
+                type: "asset/resource",
+            },
+        ],
     },
-
     resolve: {
         modules: [
-            // We need the second entry for node to be able to
-            // locate `node_modules` from client directory when 
-            // modules are referenced from inside `web-common`.
-            'node_modules', path.resolve(__dirname, './node_modules')
+            "node_modules",
+            // We need this entry for node to be able to locate `node_modules` from
+            // client directory when modules are referenced from inside `web-common`.
+            path.resolve(__dirname, "./node_modules"),
         ],
         alias: {
-            grammar: path.resolve(__dirname, './grammar.ne'),
-            static: path.resolve(__dirname, './static'),
-            src: path.resolve(__dirname, './src')
+            grammar: path.resolve(__dirname, "./grammar.ne"),
+            static: path.resolve(__dirname, "./static"),
+            src: path.resolve(__dirname, "./src")
+        },
+        extensions: [".purs", ".js", ".ts", ".tsx"],
+        fallback: {
+            "vm": require.resolve("vm-browserify"),
```

Comment:
Do we actually need this? Does this do any automatic polyfills? I wonder if it would be better to be explicit about this and not rely on this as a fallback?

I am not sure about the details behind this though.

Reply:
One of the other dependencies requires vm. In Webpack 4, polyfills for all the node stuff were included automatically, but with Webpack 5 you have to manually include stuff you need. The fallback is just a nice way getting this polyfill, since it only kicks in if you're not running in node.